subsys/storage/flash_map/CMakeLists.txt

```diff
@@ -6,6 +6,4 @@ zephyr_sources_ifdef(CONFIG_FLASH_MAP_SHELL flash_map_shell.c)
 zephyr_sources_ifdef(CONFIG_FLASH_PAGE_LAYOUT flash_map_layout.c)
 zephyr_sources_ifdef(CONFIG_FLASH_AREA_CHECK_INTEGRITY flash_map_integrity.c)
 
-if(CONFIG_FLASH_AREA_CHECK_INTEGRITY_MBEDTLS)
-  zephyr_library_link_libraries_ifdef(CONFIG_MBEDTLS mbedTLS)
-endif()
+zephyr_library_link_libraries_ifdef(CONFIG_MBEDTLS mbedTLS)
```

Comment:
Shouldn't it be guarded by `CONFIG_FLASH_AREA_CHECK_INTEGRITY_MBEDTLS` ? Wondering if there is a case where Mbed TLs together with TF-M (and consequently `FLASH_AREA_CHECK_INTEGRITY_PSA`).

Reply:
@tomi-font is on vacation so I'm inheriting this PR in the meantime.
I think he removed the check because otherwise in builds with TF-M there is this failure:
```
flash_map_integrity.c:24:10: fatal error: psa/crypto.h: No such file or directory
   24 | #include <psa/crypto.h>
```
It just needs to find the crypto header somewhere. The most "elegant" solution would be to specify:
- `mbedTLS` library when building/linking against Mbed TLS, or
- `<build-folder>/tfm/api_ns/interface/include/psa/crypto.h`

But here it comes my newbe question. In TF-M CMakeList file there is a variable named `TFM_INTERFACE_INCLUDE_DIR` which might do exactly what we're looking for, but that's not accessible from `zephyr/subsys/storage/flash_map/CMakeLists.txt`. Is there a way to make use of that to get what I described above?